samples/idTokenFromMetadataServer.js

```diff
@@ -0,0 +1,52 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * Uses the Google Cloud metadata server in the Cloud Run (or AppEngine or Kubernetes etc.,)
```

Comment:
Let's not list out the environments here. There are a number of different metadata servers, some environments use their own, some use Compute Engine's, and it's kind of complex. In the end, though, it doesn't really matter as long as the metadata server you are working with has access to your credentials. Just delete "in the Cloud Run....environment"